### PR TITLE
Warnings

### PR DESCRIPTION
**Pull request by gayanDeegala**

### CODE REVIEW

File: src/core/JsonConfigCommandCenter.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import { JSONEditorMode } from "jsoneditor";
 export class JsonConfigCommandCenter {
     public static titleUpdateCallback: (text: string, mode: JSONEditorMode) => void;
     public static getOriginalJsonConfig: () => any;
+    public static hasErrors: (error?: boolean) => boolean;
```
Comment (mihilranathunga) [inline src/core/JsonConfigCommandCenter.ts:14]: i still don't understand why this has to be a callback. why can't this be just a boolean property?

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 2]
```diff
@@ -92,90 +84,115 @@ export const CommandPanel: React.FC<{
...
+                .catch((error: any) => {
+                    message.error(LOCALIZATION.UPLOAD_ERROR.replace('{{error}}', `${extractErrorMessage(error)}`));
+                });
+        }
     }
 
     const onUpdateHandler = () => {
-        if (!isValidFileName()) {
-            message.error(LOCALIZATION.UPLOAD_INVALID_FILE);
+        if (JsonConfigCommandCenter.hasErrors()) {
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:176]: why cant hasError be a boolean property?

File: src/userInterface/panels/JsonConfigurator/JsonConfigurator.tsx [hunk 3]
```diff
@@ -26,6 +26,7 @@ export const JsonConfigurator: React.FC<any> = () => {
     const compareJsons = useRef<{ currentJson: string, newJson: string }>();
     const handleOkMerge = useRef<any>(() => { console.log('not set'); });
     const handleCancelMerge = useRef<any>(() => null);
+    let errors = false
```
Comment (mihilranathunga) [inline src/userInterface/panels/JsonConfigurator/JsonConfigurator.tsx:29]: why do we need a errors property here?

File: src/core/CogniteJsonEditorOptions.ts [hunk 4]
```diff
@@ -244,7 +244,7 @@ export class CogniteJsonEditorOptions implements JSONEditorOptions {
 
         const schemaMeta = getNodeMeta([], json).resultNode; // meta of root node from schema
         const errors = this.validateFields(json, schemaMeta);
-
+        JsonConfigCommandCenter.hasErrors(!!errors.length);
```
[the remark this answers is not in the document]
Comment (gayanDeegala) [reply src/core/CogniteJsonEditorOptions.ts]: updated

File: src/core/JsonConfigCommandCenter.ts (hunk 1 above)
Comment (gayanDeegala) [reply src/core/JsonConfigCommandCenter.ts:14]: Updated to Boolean property

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx (hunk 2 above)
Comment (gayanDeegala) [reply src/userInterface/panels/CommandPanel/CommandPanel.tsx:176]: changed

File: src/userInterface/panels/JsonConfigurator/JsonConfigurator.tsx (hunk 3 above)
Comment (gayanDeegala) [reply src/userInterface/panels/JsonConfigurator/JsonConfigurator.tsx:29]: removed

**Review by mihilranathunga — APPROVED** [review]

**Review by mihilranathunga — COMMENTED** [review]
approved

**Pull request merged**